torch/distributed/algorithms/ddp_comm_hooks/ddp_zero_hook.py

```diff
@@ -17,7 +17,7 @@ def hook_with_zero_step(
     hook: Callable[[Any, dist.GradBucket], torch.futures.Future],
     ddp: DistributedDataParallel,
     zero: ZeroRedundancyOptimizer,
-) -> Callable[[Any, dist.GradBucket], torch.futures.Future]:
+) -> Callable[[Any, dist.GradBucket], torch.futures.Future[torch.Tensor]]:
```

Comment:
I ran all of the tests via ci-all, and for a few of the tests, I saw:
```
ValueError: Communication hook: return annotation should be torch.futures.Future[torch.Tensor].
```
As a result, I changed `torch.futures.Future` -> `torch.futures.Future[torch.Tensor]`.